### PR TITLE
Add a "Report an Issue" form to catalog records

### PR DESCRIPTION
**Pull request by nicktiberi**
Re: #720.

This PR adds a "Report an Issue" button to the record toolbar. When clicked, a modal appears, showing a form that a user can fill out to report an issue. An email is then sent to `report_issue_email_recipient` (which in production + QA, should be `catalog-record-issue@PennStateOffice365.onmicrosoft.com`).

### New Button
Screen Shot 2021-04-30 at 11 28 36 AM

### Report an Issue Form/Modal
Screen Shot 2021-04-30 at 11 29 04 AM

### Form Submission Success
Screen Shot 2021-04-30 at 11 29 14 AM

### CODE REVIEW

File: config/settings.yml
Comment (nicktiberi) [inline config/settings.yml:3]: @banukutlu - I talked to @awead about this and he mentioned preferring to store info like this in environment variables which could be pulled down from Vault. Is the catalog storing any data in environment variables right now, or are all the settings just in `settings.yml`?

Not sure if this is something that needs to be changed before this PR gets merged, but wanted to bring it up.

File: app/views/catalog/_report_issue_form.html.erb
Comment (nicktiberi) [inline app/views/catalog/_report_issue_form.html.erb:1]: @awead - the linter is crapping out on this file with the error:

```
Exception occured when processing: app/views/catalog/_report_issue_form.html.erb
If this file cannot be processed by erb-lint, you can exclude it in your configuration file.
begin_pos location 2052 is out of range for document of size 2016
```

I'm not really sure why `begin_pos` is 2052 instead of 0. Have you seen something like this before?

FWIW, I replaced the contents of this file with the contents of`_email_form.html.erb` from the Blacklight gem, and it failed with the same error, but different numbers, due to the different file size.

File: config/settings.yml
Comment (awead) [reply config/settings.yml:3]: We use Vault in Scholarsphere, which just sets environment variables. I'm _assuming_ there's something similar going on here. The gist was, don't put a real email in the repo here, and the deployment (QA, prod, etc.) would set this to the real email where you want things to go.

File: app/models/record_mailer.rb
Comment (awead) [inline app/models/record_mailer.rb:20]: `_documents` isn't being used in this method. Rubocop, I bet, put that underscore in there automatically. That's a Ruby-ism for indicating that a parameter is being passed in but not used. So you should be able to delete it and just have:

``` ruby
def email_report_issue(details, url_gen_params)
...
```

File: app/controllers/catalog_controller.rb
Comment (awead) [inline app/controllers/catalog_controller.rb:65]: We're actually not using the `documents` variable here at all. When we do the `config.add_tools_partial` thing, Blacklight is generating a method called `report_issue`. That method that sets the `@documents` variable that you're using in your form. This is 👍 . But, when we make the callback to `report_issue_action`, because Blacklight has _generated_ the `report_issue` method, it's passing `@documents` into our callback. We can't change that...unless we write our own `report_issue` method, but nah... we'll just discard the documents parameter here, and only pass what the email needs:

``` ruby
def report_issue_action(_documents)                                                                                      
    mail = RecordMailer.email_report_issue(                                                                                
      title: params[:title],                                                                                               
      record_number: params[:record_number],                                                                               
      comment: params[:comment],                                                                                           
      email: params[:email],                                                                                               
      url_options: url_options)
```

This will change the method signature of `email_report_issue` slight because we're using keyword arguments:

``` ruby
def email_report_issue(title:, record_number:, comment:, email:, url_gen_params:)
```

File: app/views/catalog/_report_issue_form.html.erb
Comment (awead) [reply app/views/catalog/_report_issue_form.html.erb:1]: Weird. Yeah, I've seen this happen sometimes. I just add the file to the exclude block in `.erblint.yml`:

```
exclude:
- '**/app/views/catalog/_report_issue_form.html.erb'
```

File: spec/models/record_mailer_spec.rb
Comment (awead) [inline spec/models/record_mailer_spec.rb:11]: you shouldn't need this now.

**Review by awead — CHANGES_REQUESTED** [review]
Looks good! I hope this wasn't too annoying. In reviewing it, I think I realized that we probably would have been better off just not doing it this way to begin with, but whatever, it's working now! If you want to give it another go and do it the non-Blacklight, Rails standard way, I'm game. Otherwise, there's a few changes to get rid of `documents` and then it should be good to go.

File: app/controllers/catalog_controller.rb
Comment (nicktiberi) [inline app/controllers/catalog_controller.rb:64]: @awead - I don't love having this just chilling in `catalog_controller`. It seems like it's just cluttering the catalog controller and should maybe be in a separate file.

The main blacklight project puts `email_action` and `validate_email_params` in `app/controllers/concerns/blacklight/catalog.rb`. 

Should we do something similar? Or do you think it's fine to just leave this here? I defer to your Rails expertise. 🙂

**Comment by nicktiberi** [comment]
@awead - thanks for reviewing! If you're ok with it, I think for now maybe we can just go the Blacklight customization route. Now that I have it implemented, it does _kinda_ make sense to me. 😁 

If we need to do any more custom actions in the future, maybe we can do the standard Rails approach for the new action, and then revisit this?

File: app/controllers/catalog_controller.rb
Comment (nicktiberi) [inline app/controllers/catalog_controller.rb:85]: @awead - I don't love having this just chilling in `catalog_controller`. It seems like it's just cluttering the catalog controller and should maybe be in a separate file.

The main blacklight project puts `email_action` and `validate_email_params` in `app/controllers/concerns/blacklight/catalog.rb`. 

Should we do something similar? Or do you think it's fine to just leave this here? I defer to your Rails expertise. 🙂

Comment (awead) [reply app/controllers/catalog_controller.rb:85]: Yep. That’s a great idea. You could create a module under concerns and include it in the controller.

**Comment by nicktiberi** [comment]
@awead - should be ready for re-review. I moved those two methods into a concern and also added the record URL to the email as Ruth suggested.

**Review by awead — APPROVED** [review]

File: config/settings.yml
Comment (banukutlu) [reply config/settings.yml:3]: We are using config gem to handle these, production.local.yml or settings.local.yml where we can add the real email.

**Pull request merged**
